catalog/schematron/catalog-schematron-plugin/src/main/java/ddf/services/schematron/SchematronValidationService.java

```diff
@@ -254,39 +274,74 @@ public void setPriority(int priority) {
         }
     }
 
-    public SchematronReport getSchematronReport() {
-        return schematronReport;
-    }
-
     @Override
     public void validate(Metacard metacard) throws ValidationException {
-        try {
-            String metadata = metacard.getMetadata();
-            if (StringUtils.isEmpty(metadata) || (namespace != null
-                    && !namespace.equals(XMLUtils.getRootNamespace(metadata)))) {
-                return;
-            }
-            for (Future<Templates> validator : validators) {
-                schematronReport = generateReport(metadata, validator.get(10, TimeUnit.MINUTES));
-                if (!schematronReport.isValid(suppressWarnings)) {
-                    throw new SchematronValidationException("Schematron validation failed.",
-                            schematronReport.getErrors()
-                                    .stream()
-                                    .map(SchematronValidationService::sanitize)
-                                    .collect(Collectors.toList()),
-                            schematronReport.getWarnings()
-                                    .stream()
-                                    .map(SchematronValidationService::sanitize)
-                                    .collect(Collectors.toList()));
+
+        MetacardValidationReport report = generateReport(metacard);
+
+        List<String> errors = new ArrayList<>();
+        List<String> warnings = new ArrayList<>();
+        report.getMetacardValidationViolations()
+                .forEach(violation -> {
+                    if (violation.getSeverity() == ValidationViolation.Severity.ERROR) {
+                        errors.add(violation.getMessage());
+                    } else {
+                        warnings.add(violation.getMessage());
+                    }
+                });
+
+        SchematronValidationException exception = new SchematronValidationException(
+                "Schematron validation failed",
+                errors,
+                warnings);
+
+        if (!errors.isEmpty()) {
```

Comment:
Instead of these two `if` blocks, can you still just do a `if (!report.isValid(suppressWarnings)) {`?